### PR TITLE
chore(style): add NPM script for styling with `prettier` and run it on the codebase

### PR DESCRIPTION
**Pull request by tukusejssirs**

### CODE REVIEW

**Comment by tukusejssirs** [comment]
@emagnier, we could run `prettier` in `npm run build` too, as well as as part of `.github/workflows/dev-publishing.yml`.

**Comment by emagnier** [comment]
Not sure... I think `npm run build` should only focus on building, no code formatting.

However, GitHub Action should throw an error if the prettier checks return something that need to be prettierized..

File: package.json
Comment (emagnier) [inline package.json]: I'm not very convinced by the `style` word here. "Style" is too vague and can express anything in my point of view. Maybe just use the word `prettier`?

**Comment by tukusejssirs** [comment]
> Not sure... I think `npm run build` should only focus on building, no code formatting.

Well, I agree with you, however `prebuild` script in `package.json` is set to `npm run lint`, which is why I thought we _could_ add it in there.

However, I too think that the `build` script should only try to build, no style checking, thus no `eslint` or `prettier` should be run in there. I suggest to remove `prebuild` script altogether.

I think we should consider creating a precommit hook to run both `eslint` and `prettier`.

> However, GitHub Action should throw an error if the prettier checks return something that need to be prettierized..

We definitely should check style using both `eslint` and `prettier` before merging anything into romcal.

File: package.json
Comment (tukusejssirs) [reply package.json]: Yeah, we can change it to `prettier`, as we use to have in between versions `v1.3.1-alpha.6` and `v2.0.0-alpha.0`.

**Comment by emagnier** [comment]
Prettier should not be run when running a build (or from the prebuild script), because it will potentially modify files. When running a build, we are not supposed to modify the codebase. In addition, if the Prettier checks are failing, it should not break the build (as the contributor is probably working on something, but didn't run Prettier yet).

However, it's a good idea to:
- run Prettier in pre-commit hooks
- run Prettier:check in CI before merging anything

---

Eslint is a different topic (it's not only code styling/formating). This linter will mainly prevent problems in the codebase, analyze and detect code smell, and ensure keeping quality and codebase consistency. So having it before running the build is valuable IMO.

However, I'm not for adding eslint in pre-commit hooks. Because sometimes it's useful to create intermediate commits (while eslint issues are not all fixed yet) before submitting a PR (for example, when developing new features or doing code refactor).

File: package.json
Comment (emagnier) [inline package.json]: What about also prettify the `docs`, `scripts` and `tests` directories?

**Comment by tukusejssirs** [comment]
> Prettier should not be run when running a build (or from the prebuild script), because it will potentially modify files. When running a build, we are not supposed to modify the codebase.

Well, I actually meant running `prettier -c` and `eslint` (without `--fix`), however, that should be placed in a pre-commit (or pre-merge into `dev` branch) hook rather than as a pre-build step.

> In addition, if the Prettier checks are failing, it should not break the build (as the contributor is probably working on something, but didn't run Prettier yet).

Agreed. I think if `prettier` checks fail, it should not get merged into `dev` in the first place.

> However, it's a good idea to:
> 
>     * run Prettier in pre-commit hooks
> 
>     * run Prettier:check in CI before merging anything

I actually don’t think that fixing `prettier`/`eslint` issues should be in pre-commit/pre-merge hooks. I think it should be the committers duty. However, I am not totally against it, thus if you think we should do this automatically, I am okay with it.

> Eslint is a different topic (it's not only coding style/format).
>
> This linter will mainly prevent problems in the codebase, analyze and detect code smell, and ensure keeping quality and codebase consistency. So having it before running the build is valuable IMO.

In our current `eslint` config, it may be, however, it might check some _possible_ mistakes and errors (see my proposed `eslint` config in #383).

Still, even if there are styling issues, we should consider them blockers anyway. IMHO code style should be consistant. Contributors can style their code as they please and `eslint` should fix them all (or at least warn about them).

You seem to look at `prettier` and `eslint` as two different things, however, they have much in common. Prettier describes itself as an _opinionated code **formatter**_, while ESLint _statically analyzes code to quickly find problems_. Both fix/check styling issues, while (as I understand it) Prettier only deals with the format (style).

I personally ditched Prettier in my projects, as I kept fighting with it. I personally prefer using ESLint only. There is not one thing that Prettier does that we can’t do in ESLint (although some of the rules work differently).

> However, I'm not for adding eslint in pre-commit hooks. Because sometimes it's useful to create intermediate commits (while eslint issues are not all fixed yet) before submitting a PR (for example, when developing new features or doing code refactor).

Okay, then I suggest to create a pre-merge check (via GitHub Actions) for check the code both by `prettier` and `eslint`, no need to create a pre-commit hook at all.

Note that this is not the way I code. I try to fix all styling/formatting issues prior to creating a commit (yes, I forget to do so sometimes). However, the style/format of individual commits is not that important, only what actually merges into `dev`.

File: package.json
Comment (tukusejssirs) [reply package.json]: I have no problem with that. Do you suggest to simply update the command or update `.prettierrc`?

Comment (emagnier) [reply package.json]: Just adding that is enough, and your PR will be ready 👍 

```suggestion
    "prettier": "prettier -w --config .prettierrc {lib,docs,tests}",
    "prettier:check": "prettier -c --config .prettierrc {lib,docs,tests}",
```

Comment (tukusejssirs) [reply package.json]: Final question: should plain `prettier` script auto-fix issues or only check for issues? Accordingly, we should use `prettier:fix` or `prettier:check` for the other.

I have noticed that `lint` is just a check. We might want to be consistant in here.

Comment (emagnier) [reply package.json]: +1 for consistency with `lint`.

Let's go for `prettier` + `prettier:fix`. 
Thanks 🙏

Comment (tukusejssirs) [reply package.json]:
```suggestion
    "prettier": "prettier -c --config .prettierrc {lib,docs,tests}",
    "prettier:fix": "prettier -w --config .prettierrc {lib,docs,tests}",
```

**Comment by tukusejssirs** [comment]
@emagnier, I have resolved the issue you raised and formatted all the files using `prettier`. If it looks good to you, you could re-approve the PR. :wink:

**Review by emagnier — APPROVED** [review]

**Pull request merged**
